Remove transport client docs

rjernst opened the pull request:
This commit removes the transport client documentation.

rjernst committed:
Remove transport client docs

This commit removes the transport client documentation.

elasticmachine commented:
Pinging @elastic/es-core-infra

rjernst commented:
@lcawl I'm sure there are some cross links in other books. Should we try removing those before merging this?

lcawl commented:
When I build all docs with these changes (`./docs/build_docs --all --target_repo git@github.com:elastic/built-docs.git --sub_dir elasticsearch:master:./elasticsearch
`), I receive:

INFO:build_docs:Can't find /doc/elasticsearch/docs/java-api at lib/ES/Repo.pm line 134.

Therefore I've created https://github.com/elastic/docs/pull/900

rjernst committed:
Merge branch 'master' into remove_transport_client5

rjernst committed:
Merge branch 'master' into remove_transport_client5

rjernst committed:
Remove groovy client docs

The groovy client api was a wrapper around the transport client.
However, it has not been published since 2.4, as it had many issues with
the java security manager. This commit removes the docs from master for
the groovy client.

relates #42638

rjernst committed:
Merge branch 'remove_transport_client7' into remove_transport_client5

rjernst committed:
Remove more docs uses of transport client

rjernst committed:
Merge branch 'master' into remove_transport_client5

lcawl reviewed (approved):
LGTM and builds cleanly now
